gotham/src/router/route/matcher/accept.rs

```diff
@@ -93,30 +110,175 @@ impl RouteMatcher for AcceptHeaderRouteMatcher {
     /// matcher is only able to indicate whether a successful match has been found.
     fn is_match(&self, state: &State) -> Result<(), RouteNonMatch> {
         // Request method is valid, ensure valid Accept header
-        match HeaderMap::borrow_from(state).get(ACCEPT) {
-            // The client has not specified an `Accept` header.
-            None => Ok(()),
-
-            // Or the header is any type, so it's fine.
-            Some(header) if header == "*/*" => Ok(()),
-
-            // Otherwise we have to validate the header is a match.
-            Some(mime_header) => parse_mime_type(mime_header)
-                .map_err(|_| RouteNonMatch::new(StatusCode::NOT_ACCEPTABLE))
-                .and_then(|mime_type| {
-                    if self.supported_media_types.contains(&mime_type) {
-                        return Ok(());
-                    }
-                    trace!(
-                        "[{}] did not provide an Accept with media types supported by this Route",
-                        request_id(&state)
-                    );
-                    Err(RouteNonMatch::new(StatusCode::NOT_ACCEPTABLE))
-                }),
+
+        if self.supported_media_types.is_empty() {
+            return Ok(());
+        }
+
+        let mut headers = HeaderMap::borrow_from(state)
+            .get_all(ACCEPT)
+            .iter()
+            .peekable();
+        if headers.peek().is_none() {
+            return Ok(());
+        }
+
+        if headers.any(|hv| self.one_match(hv)) {
+            Ok(())
```

Comment:
I'd prefer if this was a `return Ok(())` to match the other statements above, and remove the `else`.